### ISSUE TITLE
pythonPackages.debugpy fails to build with Python 3

### ISSUE THREAD

**Issue by MetaDark** [NixOS/nixpkgs#104125]
**Describe the bug**
The Python 3 builds of debugpy started failing due to a change in one of the dependencies:
https://hydra.nixos.org/eval/1627728?filter=debugpy&compare=1627648&full=#tabs-still-fail.

I've started bisecting the history to try to narrow down the cause.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: python37Packages.debugpy, python38Packages.debugpy
```

**Comment by MetaDark** [comment]
I narrowed down the commit that introduced this issue to 4bfbcbeb383 `python3Packages.greenlet: 0.4.16 -> 0.4.17`. It seems like this is a bug in greenlet, but I will look into it a bit more.

**Comment by MetaDark** [comment]
Fixed by reverting greenlet back to 0.4.16: https://github.com/NixOS/nixpkgs/pull/104327
